### DISCUSSION TITLE
RISC-V-based processor (PolarFire SOC) support instruction

### DISCUSSION THREAD

**Discussion by Jpgig** [open-quantum-safe/liboqs#1352] [General]
Currently, liboqs supports x86_64 and ARM architectures.
Multi-platform. liboqs builds on Linux, macOS, and Windows, supports x86_64 and ARM architectures (except Windows on ARM64), and the clang, GCC, and Microsoft compilers. Toolchains are available for cross-compiling to other platforms.

I'm using a RISC-V-based processor(PolarFire SOC). Is there any instruction document on which files are required to change the different platforms like this?

**Comment by dstebila** [comment]
We don't have access to such a platform so we haven't tried it there, nor am I aware of anyone having done so.  If you manage to figure it out, we'd be happy to receive a CMake toolchain file or a pull request with instructions or necessary modifications.

**Comment by Jpgig** [comment]
Thanks. I can look into that but is there a document describing which files require change specific to the architecture?

**Reply by dstebila** [reply]
Not really. In principle there is pure C code for everything in the library, and CMake options to build only the pure C code, so it should be possible to quickly get such a build.

**Reply by baentsch** [reply]
@Jpgig You may want to check out https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs#oqs_dist_build (and its uses in the code to see how reference/plain C code and Platform optimized code is 'differentiated')
